scrapy/contracts/__init__.py

```diff
@@ -17,10 +17,10 @@ def __init__(self, contracts):
             self.contracts[contract.name] = contract
 
     def tested_methods_from_spidercls(self, spidercls):
+        method_re = re.compile(r"^\s*@", re.MULTILINE)
```

Comment:
a trick which works in 50% cases re.compile is used:
```suggestion
        is_method = re.compile(r"^\s*@", re.MULTILINE).search
```

Reply:
Nice, thanks!